### PR TITLE
Update UserInterface

### PR DESCRIPTION
**Pull request by XWB**
This should fix the following issues once and for all:

https://github.com/FriendsOfSymfony/FOSUserBundle/issues/1172
https://github.com/FriendsOfSymfony/FOSUserBundle/pull/1153
https://github.com/FriendsOfSymfony/FOSUserBundle/pull/1178
https://github.com/FriendsOfSymfony/FOSUserBundle/pull/1252
https://github.com/FriendsOfSymfony/FOSUserBundle/pull/1414
https://github.com/FriendsOfSymfony/FOSUserBundle/pull/1645
https://github.com/FriendsOfSymfony/FOSUserBundle/pull/1697

IMO we can easily change the interface as long as 2.0 is not final.

### CODE REVIEW

**Comment by stloyd** [comment]
This can't be merge without bumping min. PHP version to `5.3.9`.

**Comment by XWB** [comment]
Done @stloyd 
Ping @stof

File: Security/UserProvider.php [hunk 1]
```diff
@@ -56,7 +55,7 @@ public function loadUserByUsername($username)
      */
     public function refreshUser(SecurityUserInterface $user)
     {
-        if (!$user instanceof User && !$user instanceof PropelUser) {
+        if (!$user instanceof UserInterface && !$user instanceof PropelUser) {
```
Comment (xabbuh) [inline Security/UserProvider.php]: `PropelUser` implements the interface

File: Security/UserProvider.php [hunk 2]
```diff
@@ -56,7 +55,7 @@ public function loadUserByUsername($username)
...
             throw new UnsupportedUserException(sprintf('Expected an instance of FOS\UserBundle\Model\User, but got "%s".', get_class($user)));
```
Comment (xabbuh) [inline Security/UserProvider.php]: The message needs to be updated.

**Comment by XWB** [comment]
@xabbuh Done.

**Comment by stof** [comment]
> This can't be merge without bumping min. PHP version to 5.3.9.

@stloyd why ?

**Comment by stloyd** [comment]
@stof 
> __PHP documentation__: Prior to PHP 5.3.9, a class could not implement two interfaces that specified a method with the same name, since it would cause ambiguity. More recent versions of PHP allow this as long as the duplicate methods have the same signature.

**Comment by stof** [comment]
@stloyd And there is nothing in FOSUserBundle forcing you to implement a second interface containign a ``getId`` method on your class

**Pull request merged**